go/ql/src/InconsistentCode/DeferredErrorCall.ql

```diff
@@ -0,0 +1,22 @@
+/**
+ * @name Deferred call may return an error
+ * @description Deferring a call to a function which may return an error means that an error may not be handled.
+ * @kind problem
+ * @problem.severity warning
+ * @id go/deferred-error-call
+ * @tags maintainability
+ *  correctness
+ *  call
+ *  defer
+ */
+
+import go
+
+from DeferStmt defer, SignatureType sig
+where
+  // match all deferred function calls and obtain their type signatures
+  sig = defer.getCall().getCalleeExpr().getType().(SignatureType) and
+  // check that one of the results is an error
+  sig.getResultType(_).implements(Builtin::error().getType().getUnderlyingType())
+select defer,
+  "Deferred a call to " + defer.getCall().getCalleeName() + ", which may return an error."
```

Comment:
`getCalleeName` isn't always defined, so we would lose any alerts for those cases, though I can't remember exactly what they are at the moment. Is it worth trying to provide a different string when `getCalleeName` doesn't exist, to cover those cases? People defer anonymous functions quite a bit, though it would be weird to write an anonymous function with return values and then defer it.

Reply:
Fixed in https://github.com/github/codeql/pull/11410/commits/c907a9e70719cd4521a43275b5585ddd5751eae0